File: esports-backend/src/main/java/com/esports/entity/ProjectFile.java
package com.esports.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "project_files")
@Data
public class ProjectFile {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne
    @JoinColumn(name = "project_id")
    private Project project;
    
    @ManyToOne
    @JoinColumn(name = "uploader_id")
    private User uploader;
    
    @Column(nullable = false)
    private String fileName;
    
    @Column(nullable = false)
    private Long fileSize;
    
    @Column(nullable = false)
    private String s3Key;
    
    @Column(nullable = false)
    private LocalDateTime uploadedAt;
    
    @PrePersist
    protected void onCreate() {
        uploadedAt = LocalDateTime.now();
    }
}

File: esports-backend-dotnet/Upkart1/Entities/ProjectFile.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Upkart1.Entities
{
    [Table("project_files")]
    public class ProjectFile
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Column("project_id")]
        public long ProjectId { get; set; }
        public Project Project { get; set; } = null!;

        [Column("uploader_id")]
        public long UploaderId { get; set; }
        public User Uploader { get; set; } = null!;

        [Required]
        [Column("file_name")]
        public string FileName { get; set; } = null!;

        [Required]
        [Column("file_size")]
        public long FileSize { get; set; }

        [Required]
        [Column("s3_key")]
        public string S3Key { get; set; } = null!;

        [Required]
        [Column("uploaded_at")]
        public DateTime UploadedAt { get; set; } = DateTime.UtcNow;
    }
}
